python/python/lance/dataset.py

```diff
@@ -1105,6 +1105,34 @@ def delete(self, predicate: Union[str, pa.compute.Expression]):
             predicate = str(predicate)
         self._ds.delete(predicate)
 
+    def insert(
+        self,
+        data: ReaderLike,
+        *,
+        mode="append",
+        **kwargs,
+    ):
+        """
+        Insert data into the dataset.
+
+        Parameters
+        ----------
+        data_obj: Reader-like
+            The data to be written. Acceptable types are:
+            - Pandas DataFrame, Pyarrow Table, Dataset, Scanner, or RecordBatchReader
+            - Huggingface dataset
+        mode: str, default 'append'
+            The mode to use when writing the data. Options are:
+                **create** - create a new dataset (raises if uri already exists).
+                **overwrite** - create a new snapshot version
+                **append** - create a new version that is the concat of the input the
+                latest version (raises if uri does not exist)
+        **kwargs : dict, optional
+            Additional keyword arguments to pass to :func:`write_dataset`.
```

Comment:
Why call out `mode` specifically and leave `kwargs` for the rest?  Because the default changed?

Reply:
Yeah mainly for the change in default.